driver-sync/src/test/functional/com/mongodb/client/unified/EventMatcher.java

```diff
@@ -383,14 +424,53 @@ private static boolean serverDescriptionChangedEventMatches(final BsonDocument e
     }
 
     private static boolean clusterDescriptionChangedEventMatches(final BsonDocument expectedEventContents,
-            final ClusterDescriptionChangedEvent event) {
+            final ClusterDescriptionChangedEvent event, @Nullable final AssertionContext context) {
         if (!expectedEventContents.isEmpty()) {
             throw new UnsupportedOperationException(
                     "Contents of " + ClusterDescriptionChangedEvent.class.getSimpleName() + " must be empty");
         }
         return true;
     }
 
+    /**
+     * @param context Not {@code null} iff mismatch must result in an error, that is, this method works as an assertion.
```

Comment:
seems `context` in this method must be non-null, different from `serverMonitorEventMatches()` method. Seems we could delete it to avoid confusion.